airflow/models/__init__.py

```diff
@@ -703,6 +644,57 @@ def __init__(self, task, execution_date, state=None):
         # Not persisted to the database so only valid for the current process
         self.raw = False
 
+    @staticmethod
+    @provide_session
+    def clear_task_instances(tis,
+                             activate_dag_runs=True,
+                             dag=None,
+                             session=None
+                             ):
+        """
+        Clears a set of task instances, but makes sure the running ones
+        get killed.
+
+        :param tis: a list of task instances
+        :param session: current session
+        :param activate_dag_runs: flag to check for active dag run
+        :param dag: DAG object
+        """
+        job_ids = []
```

Comment:
Can we make this a set `{}`? Since the entries will be unique.

Reply:
Que?
```
C02VF05JHV2T:flink-events-ingest fdriesprong$ python3
Python 3.7.2 (default, Dec 27 2018, 07:35:06) 
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> type({1,9,2,5})
<class 'set'>
```

Reply:
```
>>> type({})
<class 'dict'>
```

empty vs with something in it

Reply:
Nice touch of magic

Reply:
Ye I know, weird stuff ;)